compiler/src/main/java/com/squareup/anvil/compiler/codegen/KotlinPoet.kt

```diff
@@ -62,7 +62,7 @@ internal fun KtCallableDeclaration.requireTypeName(
       ?: typeReference.fqNameOrNull(module)
       ?: return typeVariableName()
 
-  return ClassName.bestGuess(fqNameForGuess.asString())
+  return fqNameForGuess.asTypeName()
```

Comment:
Not obvious how this was broken or why this fixes it, were we doing something wrong or is this working around another bug?